test/extended/operators/service_catalog.go

```diff
@@ -0,0 +1,268 @@
+		upsService          = filepath.Join(buildPruningBaseDir, "ups-service.yaml")
+		upsInstance         = filepath.Join(buildPruningBaseDir, "ups-instance.yaml")
+		upsBinding          = filepath.Join(buildPruningBaseDir, "ups-binding.yaml")
+	)
+
+	enableServiceCatalogResources := []struct {
+		object    string
+		namespace string
+		name      string
+	}{
+		{"servicecatalogapiserver", "openshift-service-catalog-apiserver", "apiserver"},
+		{"servicecatalogcontrollermanager", "openshift-service-catalog-controller-manager", "controller-manager"},
+	}
+	// Enable Service Catalog
+	g.BeforeEach(func() {
+		g.By("Enable Service Catalog")
+		for _, v := range enableServiceCatalogResources {
+			_, err := oc.AsAdmin().Run("patch").Args(v.object, "cluster", "-p", `{"spec":{"managementState":"Managed"}}`, "--type=merge").Output()
+			if err != nil {
+				e2e.Failf("Unable to create: %s, error:%v", v.object, err)
+			}
+
+			err = wait.Poll(3*time.Second, serviceCatalogWait, func() (bool, error) {
+				output, err := oc.AsAdmin().Run("get").Args("pods", "-n", v.namespace, "-o=jsonpath={.items[0].status.phase}").Output()
+				if err != nil && !strings.Contains(output, "out of bounds") {
+					e2e.Failf("output string: %s", output)
+					return false, err
+				}
+				if output == "Running" {
+					e2e.Logf("%s works well!", v.object)
+					return true, nil
+				}
+				return false, nil
+			})
+
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+	}, 120)
+
+	// The below steps will cover OCP-24049, author: jiazha@redhat.com
+	// Disable these servicecatalogoperator resource
+	g.AfterEach(func() {
+		g.By("delete this ups broker")
+		output, err := oc.AsAdmin().Run("get").Args("clusterservicebroker").Output()
+		if strings.Contains(output, "ups-broker") {
+			_, err = oc.AsAdmin().Run("delete").Args("clusterservicebroker", "ups-broker").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = wait.Poll(3*time.Second, serviceCatalogWait, func() (bool, error) {
+				output, err = oc.AsAdmin().Run("get").Args("clusterserviceclass").Output()
+				if err != nil {
+					e2e.Failf("Failed to delete ups-broker, error:%v", err)
+					return false, err
+				}
+				if !strings.Contains(output, "user-provided-service") {
+					return true, nil
+				}
+				return false, nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+		for _, v := range enableServiceCatalogResources {
+			g.By("disable " + v.object)
+			_, err := oc.AsAdmin().Run("patch").Args(v.object, "cluster", "-p", `{"spec":{"managementState":"Removed"}}`, "--type=merge").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = wait.Poll(3*time.Second, serviceCatalogWait, func() (bool, error) {
+				output, err := oc.AsAdmin().Run("get").Args("ns", v.namespace).Output()
+				if err != nil {
+					if strings.Contains(output, "not found") {
+						e2e.Logf("Disable %s successfully.", v.object)
+						return true, nil
+					}
+				}
+				return false, nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
Can we make sure that there are no leftover resources when this occurs such as the `APIService`?

Reply:
Shawn is correct, after switching to `Removed` we should see if the `APIService` is still there or the namespaces for the service catalog items: `openshift-service-catalog-controller-manager` and `openshift-service-catalog-apiserver`. Though I would be okay for that to be a follow-on PR.

Reply:
@shawn-hurley @jmrodri Thanks! Sorry for the late, add the `APIservice` checking. We have already covered the namespaces checking.